content/docs/platform/inbox/configuration/preferences.mdx

```diff
@@ -0,0 +1,212 @@
+
+Each group requires:
+- **name**: The display name for the group
+- **filter**: The filter logic (object or function) to determine which workflows belong in the group
+
+There are different ways to group preferences, you can use either of them individually or combine different methods.
+
+<Callout type="info">Grouping preference is supported in client-side SDKs starting from version 3.4.0.</Callout>
+
+#### Group by tags
+
+Use an object with a `tags` key to group workflows that share specific tags.
+
+```tsx
+import { Inbox } from '@novu/react';
+
+function Novu() {
+
+  return (
+    <Inbox
+      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
+      subscriber="YOUR_SUBSCRIBER_ID"
+      preferenceGroups={[
+        {
+          name: General,
+          filter: { tags: [account] },
+        },
+      ]}
+    />
+  );
+}
+
+export default Novu;
+
+```
+
+#### Group by tags and workflow IDs
+
+Combine `tags` and `workflowIds` for more targeted grouping.
+
+```tsx
+import { Inbox } from '@novu/react';
+
+function Novu() {
+
+  return (
+    <Inbox
+      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
+      subscriber="YOUR_SUBSCRIBER_ID"
+      preferenceGroups={[
+        {
+          name: 'Marketing',
+          filter: { tags: ['marketing'], workflowIds: [workflow-ids] },
+        },
+      ]}
+    />
+  );
+}
+
+export default Novu;
+
+```
+
+#### Group all workflows
+
+Display all available workflow in a single group, regardless of its tag or grouping, by returning the full preferences list.
+
+```tsx
+import { Inbox } from '@novu/react';
+
+function Novu() {
+
+  return (
+    <Inbox
+      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
+      subscriber="YOUR_SUBSCRIBER_ID"
+      preferenceGroups={[
+        {
+          name: 'All Workflows',
+          filter: ({ preferences }) => preferences
+        },
+      ]}
+    />
+  );
+}
+
+export default Novu;
+
+```
+
+#### Group using custom logic
+
+You can define your own logic to group preferences by passing a function to the `filter` key. This function receives an object containing a `preferences` array, where each item represents a workflow preference, including both subscriber settings and workflow metadata.
+
+Each item in the `preferences` array includes a `workflow` object with properties like `name`, `slug`, and `identifier`. You can use these fields to implement your grouping logic.
+
+```tsx
+import { Inbox } from '@novu/react';
+
+function Novu() {
+
+  return (
+    <Inbox
+      applicationIdentifier="YOUR_APPLICATION_IDENTIFIER"
+      subscriber="YOUR_SUBSCRIBER_ID"
+      preferenceGroups={[
+        {
+          name: 'Alerts',
+          filter: ({ preferences }) =>
+            preferences.filter(({ workflow }) =>
+              workflow?.name?.toLowerCase().includes('alerts')
+            ),
+        },
+      ]}
+    />
+  );
+}
+
+export default Novu;
+```
+
+## Render preferences in a custom layout
```

Comment:
@Aviatorscode2 

suggestion: I feel render word is too technical, can we use something else like `display` and rethink `custom layout` word also